### ISSUE TITLE
[Bug Report]: Enbrighten/GE - ZW4002 - 55258 - 5.50

### ISSUE THREAD

**Issue by yurdle** [jascoproducts/firmware#50]
### What brand is the bug report for?

Enbrighten-GE

### ZW Number

ZW4002

### Product Number

55258

### FW Version

5.50

### Bug Report Details

Jasco [manual for ZW4002/55258](https://byjasco.com/mwdownloads/download/link/id/2563) calls out that the blue status LED will be illuminated in the dark to help you find the switch, but 55258 does not leave the LED on when the load is switched off, and there does not exist a configuration value to change the behavior.
![image](https://user-images.githubusercontent.com/5353752/173704634-784b91de-4847-4973-9255-a68542cfa12f.png)

This is reported here as well: https://community.smartthings.com/t/enbrighten-z-wave-plus-smart-fan-control-55258-led-status-light/239528

This is not a feature request, this should be considered a bug/defect as the manual calls it out as already being included and previous generations of ZW4002 worked as intended. 

Further, I personally have 5 of these installed and all behave the same way. I had one replaced under warranty and the replacement exhibit the same behavior, so I am confident this is not a hardware issue, but a firmware bug. 

### How To Reproduce

Install a new Enbrighten ZW4002/55258. 
Power on the circuit. 
Find that the status LED remains off. 
Turn the fan on, status LED flashes to indicate speed, then stops blinking, remaining off.
Turn the fan off, status LED remains off.

**Comment by cbarth3** [comment]
This is a big issue for me as every other switch in my home (20+) has the ability to have the light on when the switch is off. This needs to be added back in for this newest version which is the only version available.

**Comment by Rajamahal** [comment]
+1 to having a bunch of these switches and no configuration option for the LED. Request firmware fix

**Comment by nathan-curtis** [comment]
+1, Has this been fixed?

Just discovered this in the new Enbrighten version switches, I had an older one fail and had to replace it.  I was excited to get ZWave Plus support.  I do understand that this issue **seems** minor - but this is an issue big enough to not choose this device and instead go to another vendor.  In that case I can explain the difference in functionality.  The end user perception now is the fancy 'new, better' fan controller 'Doesn't work right.'

**Comment by jascoproducts** [comment]
Hi all.

Can't make any promises on a deadline as we haven't finished an initial version yet for testing, but development for a corrective firmware is well underway. It will be a Z-Wave Alliance-certified full release.

Thanks for your patience and sorry for the delay.

**Comment by cbarth3** [comment]
> Hi all.
> 
> Can't make any promises on a deadline as we haven't finished an initial version yet for testing, but development for a corrective firmware is well underway. It will be a Z-Wave Alliance-certified full release.
> 
> Thanks for your patience and sorry for the delay.

Just wanted to check and see if there are any updates?

**Comment by jascoproducts** [comment]
> > Hi all.
> > Can't make any promises on a deadline as we haven't finished an initial version yet for testing, but development for a corrective firmware is well underway. It will be a Z-Wave Alliance-certified full release.
> > Thanks for your patience and sorry for the delay.
> 
> Just wanted to check and see if there are any updates?

Initial firmware development is complete.

Should finish testing and be off for certification tomorrow!

**Comment by superdupe** [comment]
Can I add a +1 to this request?  Looking forward to a fix for this led issue... :)

**Comment by cbarth3** [comment]
> > > Hi all.
> > > Can't make any promises on a deadline as we haven't finished an initial version yet for testing, but development for a corrective firmware is well underway. It will be a Z-Wave Alliance-certified full release.
> > > Thanks for your patience and sorry for the delay.
> > 
> > 
> > Just wanted to check and see if there are any updates?
> 
> Initial firmware development is complete.
> 
> Should finish testing and be off for certification tomorrow!

Any more updates on possible release?

**Comment by superdupe** [comment]
I wonder what happened.  They were responding and everything was good up until about three weeks ago, and all of a sudden, radio silence... Hope whoever was the person responding on here is ok...

**Comment by jascoproducts** [comment]
@cbarth3 @superdupe 

Sorry we missed the last response to this thread.

The corrective version for this issue is still in certification which we didn't think would take so long. I am going to release a temporary beta which will be up for download indefinitely until the final version has finished certification. You will be able to use it immediately.

**Comment by superdupe** [comment]
Cool.   Thank you... :)

**Comment by superdupe** [comment]
Were you able to post the beta firmware?  I don’t see it anywhere.  Maybe I missed it?

**Comment by jascoproducts** [comment]
@superdupe should be ready tomorrow. Sorry for the delay. Will update here once it's up.

**Comment by jascoproducts** [comment]
Hi guys. We are having problems getting the beta version to update to the device properly during testing. We will try to get the version out by Monday at the latest.

**Comment by jascoproducts** [comment]
[Corrective beta firmware now available!](https://github.com/jascoproducts/firmware/tree/main/zwave/Enbrighten-GE/ZW4002/55258%20-%20In-Wall%20Fan%20Speed%20Control%2C%20500S/0.51%20(Latest%2C%20Beta))

**Comment by jascoproducts** [comment]
Closing issue: Corrective Beta firmware released. Originally this firmware version was scheduled to be officially released to mass production, however after review, no further production is scheduled for this model and product will be discontinued while we are moving to new hardware platform and updated Z-Wave 700 series.

**Comment by cbarth3** [comment]
> Closing issue: Corrective Beta firmware released. Originally this firmware version was scheduled to be officially released to mass production, however after review, no further production is scheduled for this model and product will be discontinued while we are moving to new hardware platform and updated Z-Wave 700 series.

Will this be pushed/available through home assistant like other versions?

**Comment by jascoproducts** [comment]
@cbarth3 Yes, we will be opening a pull request with them shortly to allow for updates through the HA service.

**Comment by Rajamahal** [comment]
Anyone here get this firmware to work on Home Assistant? It keeps failing for me.

**Comment by superdupe** [comment]
I don't use HA, I use Hubitat, but I do all my firmware updates by adding a USB stick with PC Controller to the hub.  Its easier to do some functions with PC Controller than with the Hubitat hub.  That being said, I was able to update all 6 of my devices to this firmware and can report it works!  The blue LED is on with the switch off...

My only issue, and not a big deal, is this firmware is named .51 instead of 5.51... Maybe because it is beta and will not ever be released as an actual update to 5.50?

**Comment by patchrick84** [comment]
Did this work for anyone else?  I used Z-Wave JS to upload it to the switch and it seems to succeed but I never get any new configuration options in the node's settings.

![image](https://user-images.githubusercontent.com/56691001/208280240-46260185-e3ff-4ea9-8641-8a892d93f548.png)

**Comment by kyleguillott** [comment]
Nope. Broke mine even more. Now my fan only works at 100%. Anything less than that and it turn off after a second or two.

**Comment by yurdle** [comment]
@patchrick84 Yes, the beta release resolved my issues. Note that this model never had the config parameter documented, so zwaveJS doesn’t expose it.
Use “Custom configuration” to manually set config parameter 3 to a value of 2 to set the LED on when the fan is off, matching the documentation for the older controller, 14314: http://manuals-backend.z-wave.info/make.php?lang=en&sku=14314&cert=ZC10-15060022&type=mini

**Comment by Rajamahal** [comment]
@patchrick84 Yes eventually but it took many update attempts per switch through home assistant to get it to 100% completion without freezing up and having to remove node, add, try again.

**Comment by meluvalli** [comment]
I am also running HomeAssistant.  I am not sure how to configure this manually.  I updated to the firmware 0.51 successfully, and HA shows I am on that firmware for the device, but I can't change the parameter.  I tried running the service manually with the following:
<code>
service: zwave_js.set_config_parameter
data:
  parameter: "3"
  value: "2"
target:
  entity_id: fan.in_wall_3_speed_fan_control
</code>
However, it just fails.  And doesn't change anything.  Any suggestions?

**Comment by Rajamahal** [comment]
> I am also running HomeAssistant. I am not sure how to configure this manually. I updated to the firmware 0.51 successfully, and HA shows I am on that firmware for the device, but I can't change the parameter. I tried running the service manually with the following:
> ` service: zwave_js.set_config_parameter data: parameter: "3" value: "2" target: entity_id: fan.in_wall_3_speed_fan_control `
> However, it just fails. And doesn't change anything. Any suggestions?

![A8C8BCC4-FB3E-4D8E-9E9B-A248C4E4AD4D](https://user-images.githubusercontent.com/98786619/208698468-937e41d8-a43e-466e-9d8e-6661033c9c95.jpeg)

**Comment by jascoproducts** [comment]
Reopening this as an ongoing discussion and to make room for a new pin.

**Comment by meluvalli** [comment]
> > I am also running HomeAssistant. I am not sure how to configure this manually. I updated to the firmware 0.51 successfully, and HA shows I am on that firmware for the device, but I can't change the parameter. I tried running the service manually with the following:
> > `service: zwave_js.set_config_parameter data: parameter: "3" value: "2" target: entity_id: fan.in_wall_3_speed_fan_control`
> > However, it just fails. And doesn't change anything. Any suggestions?
> 
> ![A8C8BCC4-FB3E-4D8E-9E9B-A248C4E4AD4D](https://user-images.githubusercontent.com/98786619/208698468-937e41d8-a43e-466e-9d8e-6661033c9c95.jpeg) @meluvalli

Thank you!  I installed and configured Z-Wave JS UI and this worked.  I am not sure why I couldn't do it without the UI with the commands I sent before, but this did the trick!  Thanks again!

**Comment by patchrick84** [comment]
> ![A8C8BCC4-FB3E-4D8E-9E9B-A248C4E4AD4D](https://user-images.githubusercontent.com/98786619/208698468-937e41d8-a43e-466e-9d8e-6661033c9c95.jpeg) @meluvalli

This also worked for me, although I needed to use a value of 1 instead of 2 for some reason.  Thank you for the hint!

**Comment by jascoproducts** [comment]
Cleaning up issues - closing this issue as resolved.

**Comment by clarkja5** [comment]
I forgot to mention I had to do a manual update with the firmware from the GitHub repository it wouldn't work using ota updates.

**Comment by clarkja5** [comment]
I have both because at first I thought maybe the new firmware was the problem, but I think I used the one in the zip. I'm just sitting down to really look at it since that update. I am running Zwave JS Ui running in HA. It is currently showing FW: v0.51 but it is giving me the option to upgrade to 5.51. I think I'm going to let it try and see what happens.

**Comment by steverep** [comment]
> I just went into devices in HA and the firmware listed there is v5.51

This doesn't match your previous comment, and it's not what I see.

**Comment by clarkja5** [comment]
I have taken another look and I see where I saw v5.51. It shows v0.51 in the Zwave JS UI node list and in HA devices under device info, but if I scroll down to the Configuration box that is where it says Zwave firmware is 5.51

**Comment by nmpu** [comment]
The versioning issue has been resolved. Jasco had simply renamed the file without updating the internal version.

@jascoproducts Let's suppose parameter 40 is set to 1: _Press the paddle up **or** down **once** to cycle between fan speeds_.

I understand how tap up could work. In fact, I already have this implemented using the raw button events. I have it coded so that tapping up _while already on_ will cycle low->medium->high->low. I also have it coded so that tapping down _when already off_ will start on low. In all other cases, the default behaviors are maintained-- tapping down is instant off and tapping up resumes at the previous value.

 How is your tap down supposed to work? If the fan is on high, do we need to tap down 3 times to turn off?
